### ISSUE TITLE
Binary link dead

### ISSUE THREAD

**Issue by OMendozaG** [OSVR/SteamVR-OSVR#136]
Hi,

The link to the binary files is dead. (Someone has bought the domain and redirected it).

On reddit there are a lot of people asking for the binaries.
OSVR works and my glasses still working.

They are useful for some things and I don't want to let them die just because I can't connect them with SteamVR which are very useful for testing VR in Unity.

Compiling the binaries with the instructions is impossible at least for me. I'm looking for Windows x64.

Could someone who keeps them share them with all of us who are looking for them?
Or maybe someone who knows how to do it could compile and upload them?

I don't know if the new version of Steam will be compatible but surely it will and just because the link is dead many of us are stuck without being able to use our OSVR.

Please, could you help me?

**Comment by russell-taylor** [comment]
Does this file work?
https://drive.google.com/file/d/175cmSxqUIkEtEirWkWnzlKbbJnMCxPbO/view?usp=sharing

**Comment by OMendozaG** [comment]
thanks!!

**Comment by OMendozaG** [comment]
Didn't work :( maybe changes in steam VR , old compilation or some files missing in dir tree? Can you help me please?

I have HDK2, does it continue to work for you? @russell-taylor

**Comment by russell-taylor** [comment]
I haven't tried using my HDK2 in awhile, and I didn't usually use it on SteamVR.  That was a Zip file I had squirreled away from another devloper's build.

I recommend trying the Wayback Machine to see if you can find a cached link, or see if someone on Reddit has a copy that they saved.  If so, please post it here.

**Comment by kungfooman** [comment]
Any news?

I also just tried to build it (+ Core, boost, libfunctionality, json lib and so on), but it's a C++ dependency hell, spent nearly two days now without really knowing what I need to make it work.

Maybe someone still has binaries around?

For example I'm searching:

https://www.reddit.com/r/OSVR/comments/5yhjnn/hdk_windows_installer_beta_070_released/
https://www.reddit.com/r/OSVR/comments/692hgo/new_release_of_steamvrosvr_build_342/
https://www.reddit.com/r/OSVR/comments/67hqrf/hdk_windows_installer_beta_080_released/

**Comment by rpavlik** [comment]
See https://github.com/OSVR/OSVR-Core/issues/606#issuecomment-1917708707 - I do have the data, it should be a lot easier to build nowadays with vcpkg but you wouldn't have the NDA parts for direct mode anyway.

**Comment by Conzar** [comment]
You might want to checkout https://monado.freedesktop.org/ as it supports the HDK2.

**Comment by kungfooman** [comment]
> You might want to checkout https://monado.freedesktop.org/ as it supports the HDK2.

Thank you for the suggestion, I just tried it. I installed monado via `apt` and then tried to start `monado-service`, but it errors on `does not support required extension VK_KHR_external_semaphore_fd`.

Then I realized that I need OSVR-Core in combination with Monado, but I tried building OSVR-Core once before and Linux and it failed, because of some GCC C++ intricacies (C++ projects always break after a few years...).

Did you compile OSVR-Core yourself and got monado to work with SteamVR?

**Comment by Conzar** [comment]
I am actively working on [my own branch of monado](https://gitlab.freedesktop.org/mspeth/monado/-/commits/nolo-standalone) to include NOLOVR (to enable positional tracking to HDK2).  Monado only supports rotational tracking for the HDK2 as of now (I never was able to get the IR tracking working due to the proprietary hardware/software in the IR tracker that was never open sourced).

OSVR-Core isn't required for Monado.  You should be able to just build Monado from source.  Checkout the [getting started](https://gitlab.freedesktop.org/monado/monado#getting-started) to ensure you install the required dependencies.

I use the following script to build monado. 

```
#!/bin/bash
ROOT=<PATH_TO_MONADO_SOURCE>/monado
TARGET=/opt/monado
pushd .
cd $ROOT
cmake -G Ninja -S . -B build -DCMAKE_INSTALL_PREFIX=$TARGET -DCMAKE_BUILD_TYPE=RelWithDebInfo  -DCXRT_BUILD_DRIVER_OHMD=false
ninja -C build install
popd
```

For additional help with monado, please[ join the discord](https://discord.gg/8RkJgRJ).  There are a lot of developers and users that are willing to help!

**Comment by kungfooman** [comment]
I got HDK2 to work on Windows 10 with latest SteamVR version now - I'm so glad about it!

The missing piece was this SteamVR driver: https://archive.org/details/steamvr-osvr

I tried an older one before I don't even remember where I found it and it simply didn't work.

I drove this track in Assetto Corsa - Trento Bondone: https://www.youtube.com/watch?v=rddH8sxegoU

It took me 23 minute and it worked out quite nicely (only IR camera tracking doesn't work so far).

**Comment by rpavlik** [comment]
yeah, no, Monado won't help you run Assetto Corsa, that's an OSVR game I think. (Or very old steamvr?) I'm glad you found some on archive.org - that was one of the options I was weighing for where to put the accumulated builds, since I assume Razer will eventually stop paying that S3 bill. If anyone has experience uploading stuff there, please get in touch, I'd love to get this stuff preserved in an easy to find location with good metadata.

The only non-open-source firmware in the IR tracker is in the camera, and that's because they negotiated everything with the camera vendor without looping me in: I was handed the final camera and not involved before. Fortunately there is a firmware update (requested by me and my team) that fixes the USB descriptors so it works on Linux. Everything else should work on Linux fine, though I haven't tried building it in a few years (as you might imagine, given that Sensics folded and I work elsewhere now - fortunately still on open source VR stuff.). To be honest, a pass with clang-tidy and some "modernize" settings should be able to migrate nearly all of the boost dependencies away to the C++17/20 standard library. OpenCV for the camera tracker is probably the hardest dep to keep up with.

**Comment by Conzar** [comment]
> The only non-open-source firmware in the IR tracker is in the camera, and that's because they negotiated everything with the camera vendor without looping me in: I was handed the final camera and not involved before. Fortunately there is a firmware update (requested by me and my team) that fixes the USB descriptors so it works on Linux. Everything else should work on Linux fine

I never was able to get the IR camera working on Linux.  I haven't tried in years though.

**Comment by Nandox7** [comment]
> https://www.reddit.com/r/OSVR/comments/5yhjnn/hdk_windows_installer_beta_070_released/
> https://www.reddit.com/r/OSVR/comments/692hgo/new_release_of_steamvrosvr_build_342/
> https://www.reddit.com/r/OSVR/comments/67hqrf/hdk_windows_installer_beta_080_released/

I'll check in my disks. I have the HDK2 + NOLOVR, stopped using as my GPU was to weak.

Me and two other friends got them at the same time, so between the three of us we might still have those files somewhere.
